web/ui/src/features/river-js/RiverValue.tsx

```diff
@@ -80,19 +115,25 @@ const ValueRenderer: FC<valueRendererProps> = (props) => {
             // partition.
             const keyLength = partitionKeyLength(partition);
 
-            return partition.map((element, index) => {
-              return (
-                <Fragment key={index.toString()}>
-                  {getLinePrefix(props.indentLevel + 1)}
-                  <span>{partitionKey(element, keyLength)} = </span>
-                  <ValueRenderer value={element.value} indentLevel={props.indentLevel + 1} />
-                  <span>,</span>
-                  <br />
-                </Fragment>
-              );
+            // Loop through the partition and return a simpler object with key and value pair
+            const simplifiedPartition = partition.map(({ key, value }) => {
+              return {
+                key,
+                value: typeof value === 'object' && 'value' in value && (value.value as string),
+              };
             });
+
+            const gridTemplateColumns = '10% 1% 69%';
+
+            switch (name) {
+              case 'targets':
+                return renderGrid(simplifiedPartition);
+              case 'labels':
+                return renderGrid(simplifiedPartition, gridTemplateColumns);
+              default:
+                return renderDefault(partition, indentLevel, keyLength);
+            }
```

Comment:
> Unfortunately this won't work in the general case. We can't guarantee that the only time a object or array value will be in attributes called `targets` and `labels`.

Thanks for the context. In this case, we can apply just the `renderGrid()` without the condition, and combine the `renderGrid` with `renderDefault`, which by adding the `<ValueRenderer>` in the `renderGrid` to fix the out of bound at the same time not breaking the layout. This way it will still call the recursive and detect the `ValueType` for the value. I will do a small adjustment on this.

After the combination it will still detect the color since we will not be using the `grid-value` color anymore. The only thing is to add a dynamic calculation on the width of the css.